drivers/flash/flash_simulator.c

```diff
@@ -148,7 +151,10 @@ static bool flash_erase_at_start;
 static bool flash_rm_at_exit;
 static bool flash_in_ram;
 #else
-#if DT_NODE_HAS_PROP(DT_PARENT(SOC_NV_FLASH_NODE), memory_region)
+#ifdef CONFIG_FLASH_SIMULATOR_SEMIHOST
```

Comment:
No need to nest ifs
```suggestion
#elif defined(CONFIG_FLASH_SIMULATOR_SEMIHOST)
```
(and remove the last #endif)